### PR TITLE
Raise when task is cancelled or timed out

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR fixes timeout tests on python >= 3.9

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by angrybayblade** [comment]
Marco has already explained the changes on https://github.com/valory-xyz/open-aea/pull/166 so I suppose there's no point in me explaining the changes at this point

**Comment by angrybayblade** [comment]
The failures on `tests/test_manager/test_manager.py::TestMultiAgentManagerMultiprocessMode` on windows 10, python 3.10 is caused by this particular method `install_pypi_dependencies` which is supposed to be patched 

https://github.com/valory-xyz/open-aea/blob/1d21aa09430ab4f31d9036ebd791312845b5dd4f/tests/test_manager/test_manager.py#L687

**Review by DavidMinarsch — APPROVED** [review]

File: aea/helpers/async_utils.py [hunk 1]
```diff
@@ -463,8 +463,11 @@ async def _run_wrapper(self) -> None:
...
         self._is_running = True
         try:
-            with suppress(asyncio.CancelledError):
+            try:
                 return await self.run()
+            except asyncio.CancelledError:
+                if not self._was_cancelled:
+                    raise
         finally:
             self._is_running = False
```
Comment (DavidMinarsch) [inline aea/helpers/async_utils.py:472]: what about these diffs on @marcofavorito's version? https://github.com/valory-xyz/open-aea/pull/166/files#diff-870febf297e27b2d3bcc0cf64e15bb69aff4f276146252956f669d2c101f6e4bR505-R572

File: tests/test_packages/test_connections/test_stub/test_stub.py [hunk 2]
```diff
@@ -351,10 +350,6 @@ async def wait_num(num):
 
 @pytest.mark.flaky(reruns=MAX_FLAKY_RERUNS)
 @pytest.mark.asyncio
-@pytest.mark.skipif(
-    sys.version_info.major == 3 and sys.version_info.minor == 9,
-    reason="needs investigation; repeatedly fails on py3.9",
-)  # needs investigation; repeatedly fails on py3.9
 async def test_bad_envelope():
```
Comment (DavidMinarsch) [inline tests/test_packages/test_connections/test_stub/test_stub.py:353]: Check marco's edits too: https://github.com/valory-xyz/open-aea/pull/166/files#diff-97f41f25d8a8b22de6b962f24b394451a7e4739d88369e833d8e9a4c2820d6b8R370-R371

**Comment by Karrenbelt** [comment]
obvious issues are:
- one fail on ubuntu with 3.10 
    `FAILED tests/test_manager/test_manager.py::TestMultiAgentManagerMultiprocessMode::test_start_all`
- and 20 on windows with 3.10. All relate to the same test
`FAILED tests/test_manager/test_manager.py::TestMultiAgentManagerMultiprocessMode`

I looked into the logs a bit and find this

```
aea.exceptions.AEAEnforceError: pip install failed. Return code != 0: stderr is b'ERROR: 
Cannot install open-aea-ledger-fetchai because these package versions have conflicting dependencies.\r\nERROR: 
ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts\r\n'
```

**Comment by angrybayblade** [comment]
> obvious issues are:
> 
> * one fail on ubuntu with 3.10
>   `FAILED tests/test_manager/test_manager.py::TestMultiAgentManagerMultiprocessMode::test_start_all`
> * and 20 on windows with 3.10. All relate to the same test
>   `FAILED tests/test_manager/test_manager.py::TestMultiAgentManagerMultiprocessMode`
> 
> I looked into the logs a bit and find this on the ubuntu run
> 
> ```
> aea.exceptions.AEAEnforceError: pip install failed. Return code != 0: stderr is b'ERROR: 
> Cannot install open-aea-ledger-fetchai because these package versions have conflicting dependencies.\r\nERROR: 
> ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts\r\n'
> ```

https://github.com/valory-xyz/open-aea/pull/165#issuecomment-1147083507

**Pull request merged**